src/types.ts

```diff
@@ -0,0 +1,72 @@
+const { Context } = require("@distributedlab/circom2");
+
+export type NumericString = `${number}` | string;
+
+export type PublicSignals = NumericString[];
+
+export type Groth16Proof = {
+  pi_a: [NumericString, NumericString];
+  pi_b: [[NumericString, NumericString], [NumericString, NumericString]];
+  pi_c: [NumericString, NumericString];
+  protocol: string;
+  curve: string;
+};
+
+export type Calldata = [
+  [NumericString, NumericString],
+  [[NumericString, NumericString], [NumericString, NumericString]],
+  [NumericString, NumericString],
+  [NumericString],
+];
+
+export type ProofStruct = {
+  proof: Groth16Proof;
+  publicSignals: PublicSignals;
+};
+
+export type NumberLike = number | bigint | string;
+
+export type Inputs = Record<string, NumberLike>;
+
+export type CircuitZKitConfig = {
+  circuitFile: string;
+  artifactsDir: string;
+  circuitOutDir: string;
+  verifierOutDir: string;
+};
+
+export type CircomZKitConfig = {
+  circuitsDir?: string;
+  artifactsDir?: string;
+  verifiersDir?: string;
+};
+
+export type CircomZKitPrivateConfig = {
+  circuitsDir: string;
+  artifactsDir: string;
+  verifiersDir: string;
+};
+
+export type CircuitZKitPrivateConfig = {
```

Comment:
`ptauFile`, `ptauDir`, `groth16Template`, etc. are global values, not specific to the circuit, IMO, it is better to separate them